### ISSUE TITLE
single end dataset collection as input of Trinity

### ISSUE THREAD

**Issue by drosofff** [galaxyproject/tools-iuc#1704]
Dear trinity maintainers,

I have doubts that https://github.com/galaxyproject/tools-iuc/blob/master/tools/trinity/trinity.xml#L53 is sufficient to handle a collection of single end read datasets as input of the trinity tool.

Can you help on this ?

**Comment by abretaud** [comment]
I've just tested, it seems to work as expected. What exactly is your problem?

**Comment by drosofff** [comment]
Hi Antony,
Sorry I did not see you reply.

The tool does not have a test for a collection of *single end* reads.
When I test this setting, it fails. I though that the  https://github.com/galaxyproject/tools-iuc/blob/master/tools/trinity/trinity.xml#L53 may explain this bug.

I can give you access to a fresh galaxy instance, but this is a GCE VM so I need to start it up before. Ask me if you want to have a look (but testing a single end read collection locally should in principle show you the same problem)

**Comment by abretaud** [comment]
I don't see the problem, at https://github.com/galaxyproject/tools-iuc/blob/master/tools/trinity/trinity.xml#L54 we loop on all the single end datasets provided. As the param is `multiple="true"`, the user can select multiple datasets or collections and run trinity on it.

**Comment by mvdbeek** [comment]
> multiple="true", the user can select multiple datasets or collections and run trinity on it.

`multiple="true"` is a reduction operation, so this only makes sense if the samples you select there are supposed to be merged and treated as one. It probably makes more sense to either remove `multiple="true"` or wrap the merging in a conditional.

**Comment by abretaud** [comment]
> merged and treated as one

That's the intended beaviour, it is an assembler which takes all available fastq to produce a single result

**Comment by mvdbeek** [comment]
I wouldn't call that expected, maybe for metagenomics it makes sense, but for a multi-sample analysis where you have one fastq per sample this is not what you'd expect. We've had that discussion a few times before and I think until the galaxy ui lets you specify the behaviour we should wrap this in aconditional, as we've done for macs2 recently (https://github.com/galaxyproject/tools-iuc/pull/1648)

**Comment by abretaud** [comment]
You mean https://github.com/galaxyproject/tools-iuc/pull/1648/files#diff-77d32f534e7d91585bd3f97991dbd623R127 ?

For trinity I don't think it makes sense to add a conditional like "Are you pooling Treatment Files?", you always want to pool the input files

**Comment by mvdbeek** [comment]
Yep, that's the one

> For trinity I don't think it makes sense to add a conditional like "Are you pooling Treatment Files?", you always want to pool the input files

I don't think we should be dictating this. If you have individual, independent samples that you want to assemble to survey for unknown / unexpected transcripts you really don't want to merge them.

**Comment by mvdbeek** [comment]
@drosofff is that the problem you were having, a collection being collapsed to a single output ?

**Comment by drosofff** [comment]
@mdvbeek @abretaud Absolutely, I am using trinity in parallel de novo assemblies and pooling the collection datasets is not appropriate in this configuration.
This said, sorry to insist, but there is not test for single-end dataset collection in the xml.  And in addition to the true issue discussed above, I have an error - red state - on my side. I am trying to understand whether it comes from my environment or from the tool.
